AutomatedLabWorker/AutomatedLabAzureWorkerVirtualMachines.psm1

```diff
@@ -1242,50 +1232,33 @@ catch
 }
 '@
     $customScriptContent | Out-File $tempFileName -Force -Encoding utf8
-    $null = Set-AzureStorageBlobContent -File $tempFileName -Container labsources -Blob Enable-WinRm.ps1 -BlobType Block -Context $storageAccount.Context -Force
-    Remove-Item $tempFileName -Force -ErrorAction SilentlyContinue   
+    $rgName = Get-LabAzureDefaultResourceGroup
 
-    foreach ($m in $Machine)
+    $jobs = foreach ($m in $Machine)
     {
-        $jobs += Start-Job -Name "AzureRemotingActivation ($($m.Name))" -ScriptBlock {
-            param
-            (
-                $MachineName,
-                $ResourceGroup,
-                $Location,
-                $AzureRetryCount,
-                $StorageAccountName,
-                $StorageAccountKey
-            )
-            
-            $i = 0
-            while (-not $vmExtension.IsSuccessStatusCode -and $i -le $AzureRetryCount)
-            {
-                $vmExtension = Set-AzureRmVMCustomScriptExtension -VMName $MachineName -ContainerName 'labsources' `
-                -FileName 'Enable-WinRm.ps1' -StorageAccountName $StorageAccountName `
-                -StorageAccountKey $StorageAccountKey -Run Enable-WinRm.ps1 `
-                -ResourceGroupName $ResourceGroup -Name WinrmActivation -Location $Location -Verbose -ErrorAction SilentlyContinue
-                $i++
-                Start-Sleep -Seconds 1
-            }
-
-            if (-not $vmExtension -or -not $vmExtension.IsSuccessStatusCode)
-            {
-                throw "Setting up WinRm on $machineName failed!"
-            }
-        } -ArgumentList $m.Name, (Get-LabAzureDefaultResourceGroup), (Get-LabAzureDefaultLocation), $azureRetryCount, $storageAccountName, $storageAccountKey
+        Invoke-AzureRmVMRunCommand -ResourceGroupName $rgName -VMName $m.Name -ScriptPath $tempFileName -CommandId 'EnableLabRemoting' -ErrorAction Stop -AsJob
```

Comment:
Please change the CommandId to RunPowerShellScript. Then this PR is ready to merge.

Reply:
I had. I don't know where that went